### ISSUE TITLE
Update xlc compiler version

### ISSUE THREAD

**Issue by keithc-ca** [eclipse-openj9/openj9#19098]
The minimum was changed to 17.1.1.4 in https://github.com/ibmruntimes/openj9-openjdk-jdk/commit/b0474fc291128169f16db15b6cf616a59568cec5. A warning is now generated at configuration time - see https://openj9-jenkins.osuosl.org/job/Build_JDKnext_ppc64_aix_OpenJDK/633:
```
[2024-03-07T05:41:51.867Z] configure: WARNING: You are using xlc 16.01.0000.0011 which is older than 17.1.1.4. This is not a supported configuration.
```

**Comment by pshipton** [comment]
Yes we should update the head stream. Afaik xlc 17 options aren't compatible with xlc 16 options so we can't just switch. We need to add support to OpenJ9/OMR to use the correct option syntax.

**Comment by keithc-ca** [comment]
The options can't be too different or the build would have failed (if openjdk used xlc 17 options that xlc 16 doesn't understand), although it's possible that OpenJ9 or OMR use options that won't work as is with xlc 17.

**Comment by pshipton** [comment]
OpenJDK added support for xlc 17, if we tried to do a build using it OpenJDK would detect xlc 17 and do the appropriate thing.

**Comment by keithc-ca** [comment]
I don't see anything that's sensitive to the version of xlc (other than the warning that 16.1 is unsupported), got an example?

**Comment by pshipton** [comment]
I think it was a misunderstanding on my part. Looking back when I asked for xlC 17 to be installed on a machine, it seems I only got ibm-clang, which uses different options. At the time I must have thought that was the only option. I've again requested getting xlc 17 installed in order to give it a try.

**Comment by pshipton** [comment]
There is https://github.com/ibmruntimes/openj9-openjdk-jdk22/commit/2e9cfe2d806

**Comment by keithc-ca** [comment]
Yes `ibm-clang` is very different from `xlc` (which openjdk mistakenly refers to as ["Open XL"](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/openj9/doc/building.md#native-compiler-toolchain-requirements)).

**Comment by pshipton** [comment]
ibm-clang comes in a directory called openxlC. I'm not yet 100% sure there is anything else. 
https://www.ibm.com/products/open-xl-cpp-aix-compiler-power describes Open XL and links to "classic compilers" https://www.ibm.com/docs/en/xl-c-and-cpp-aix which only includes up to 16.

**Comment by keithc-ca** [comment]
It's my understanding that `ibm-clang` is the name of an executable in "Open XL"; IBM doesn't refer to the toolchain that includes `xlc` as "Open" (even though openjdk does) - that's the "classic" toolchain.

**Comment by pshipton** [comment]
Checking with Julian about 17, "macro(s) and options are not compatible anymore". There is an ongoing effort to get OpenJ9 working.

**Comment by pshipton** [comment]
We could try removing the error to see if it works anyway.

**Comment by keithc-ca** [comment]
I started an internal build with [8327701: Remove the xlc toolchain](https://github.com/ibmruntimes/openj9-openjdk-jdk/commit/ceeb8571963012d55755ad7d875b5fedb93e7db1) reverted. It has gotten past the configuration issue. I'll create a pull request for that change if all goes well.

**Comment by keithc-ca** [comment]
Internally, it didn't go as smoothly as I had hoped: trying externally via https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/757.

**Comment by keithc-ca** [comment]
We need not delay use of OpenXL 17 until OpenJ9 and OMR can build with that compiler; jdknext can be configured using `--with-openj9-cc` and `--with-openj9-cxx` so OpenJ9 and OMR can continue to use xlc.

**Comment by pshipton** [comment]
Should we be doing that rather than merging https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/757 ?

**Comment by pshipton** [comment]
To answer my own question, we don't have machines ready to support using Open XL 17.

**Comment by keithc-ca** [comment]
Even if we did have suitable machines, I wouldn't expect them to be updated immediately, so I suggest https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/757 be merged for now; we can revert those changes when we have Open XL 17 installed.

**Comment by pshipton** [comment]
With https://github.com/eclipse-openj9/openj9/pull/23070 and the backports of https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/1136, this is completed.
